File: UI/src-tauri/src/models/request/request_subscription.rs
// Subscription / payment request payloads.
use serde::Deserialize;

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RequestCreateOrder {
    pub user_id: String,
    pub plan_id: String,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RequestVerifyPayment {
    pub razorpay_order_id: String,
    pub razorpay_payment_id: String,
    pub razorpay_signature: String,
    pub user_id: String,
    pub plan_id: String,
}

File: UI/src/app/models/request/requestSubscription.ts
// Subscription / payment request payloads — mirrors
// UI/src-tauri/src/models/request/request_subscription.rs.

export interface requestCreateOrder {
  userId: string;
  planId: string;
}

export interface requestVerifyPayment {
  razorpayOrderId: string;
  razorpayPaymentId: string;
  razorpaySignature: string;
  userId: string;
  planId: string;
}
